xapian-core/tests/harness/backendmanager_multi.cc

```diff
@@ -91,28 +113,44 @@ BackendManagerMulti::createdb_multi(const string& name,
     // Open NUMBER_OF_SUB_DBS databases and index files to them alternately so
     // a multi-db combining them contains the documents in the expected order.
     Xapian::WritableDatabase dbs;
-    const string& subtype = sub_manager->get_dbtype();
-    int flags = Xapian::DB_CREATE_OR_OVERWRITE;
-    if (subtype == "glass") {
-	flags |= Xapian::DB_BACKEND_GLASS;
-    } else {
-	string msg = "Unknown multidb subtype: ";
-	msg += subtype;
-	throw msg;
-    }
+
     string dbbase = db_path;
     dbbase += "___";
     size_t dbbase_len = dbbase.size();
-    string line = subtype;
-    line += ' ';
-    line += dbname;
-    line += "___";
+
     for (size_t n = 0; n < NUMBER_OF_SUB_DBS; ++n) {
-	dbbase += str(n);
-	dbs.add_database(Xapian::WritableDatabase(dbbase, flags));
+	const string &subtype = sub_managers[n]->get_dbtype();
+	int flags = Xapian::DB_CREATE_OR_OVERWRITE;
+	if (subtype == "glass") {
+	    flags |= Xapian::DB_BACKEND_GLASS;
+	    dbbase += str(n);
+	    dbs.add_database(Xapian::WritableDatabase(dbbase, flags));
+	    string line = subtype + " " + dbname + "___" + str(n);
+	    out << line << '\n';
+	} else if (subtype == "remoteprog_glass") {
+	    flags |= Xapian::DB_BACKEND_GLASS;
+	    dbbase += str(n);
+	    Xapian::WritableDatabase remote_db(dbbase, flags);
+	    remote_db.close();
+	    string args = sub_managers[n]->get_writable_database_args(dbbase, 300000);
+#ifdef HAVE_VALGRIND
+	    if (RUNNING_ON_VALGRIND) {
+		args.insert(0, XAPIAN_PROGSRV" ");
+		dbs.add_database(Xapian::Remote::open_writable("./runsrv", args));
+		out << "remote :" << args << '\n';
+	    }
+#else
+	    dbs.add_database(Xapian::Remote::open_writable(XAPIAN_PROGSRV, args));
+	    out << "remote :" << XAPIAN_PROGSRV << " " << args << '\n';
+#endif
```

Comment:
This seems like way too much knowledge about remote databases to be here rather than in `BackendManagerRemote`.